### ISSUE TITLE
39c41e3 breaks REMIND input data processing via readGCPT()

### ISSUE THREAD

**Issue by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q** [pik-piam/mrremind#40]
https://github.com/pik-piam/mrremind/blob/df518845e05936b742e1dcbe3b295eb43f56fad2/R/readGCPT.R#L125

Yeah, no.  I can tell you it breaks down there
```R
> mavgRetAge[which(mavgRetAge==0)]

  [1] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 [47] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 [93] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
[139] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
[185] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
```
but you will have to figure out how to do this.  I don't touch `magclass` with a 10' pole, if I can avoid it.

Also, to save you and me the trouble of another issue:

https://github.com/pik-piam/mrremind/blob/df518845e05936b742e1dcbe3b295eb43f56fad2/R/readGCPT.R#L190-L193

```
$ ll /p/projects/rd3mod/inputdata/mappings/regional/regionmappingPPCA.csv \
/p/projects/rd3mod/inputdata/mappings/regional/regionmappingOECD.csv
ls: cannot access '/p/projects/rd3mod/inputdata/mappings/regional/regionmappingPPCA.csv': No such file or directory
ls: cannot access '/p/projects/rd3mod/inputdata/mappings/regional/regionmappingOECD.csv': No such file or directory
```

**Comment by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q** [comment]
> but you will have to figure out how to do this.  I don't touch `magclass` with a 10' pole, if I can avoid it.

Had a 11' epiphany.  `getRegions(mavgRetAge)[mavgRetAge == 0]` is what you want.
```
> all(mavgRetAge[getRegions(mavgRetAge)[mavgRetAge==0],,] == 0)
[1] TRUE
> all(mavgRetAge[getRegions(mavgRetAge)[mavgRetAge==0],, invert = TRUE] != 0)
[1] TRUE
```

**Comment by stephenbi** [comment]
I uploaded the mapping files to address part 2 of the issue.

As for part 1, the two implementations are equivalent (so I don't get a bug, at least on my machine). Do you mean that my syntax is unorthodox/unstable?

**Comment by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q** [comment]
What `magclass` version are you running?

**Comment by stephenbi** [comment]
5.13.1... looks like I'm a month out of date

**Comment by stephenbi** [comment]
Fixed in 71927f0c7dcc84ece7a819763ea639c88a2f7f88

**Comment by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q** [comment]
> Fixed in [71927f0](https://github.com/pik-piam/mrremind/commit/71927f0c7dcc84ece7a819763ea639c88a2f7f88)

Wrong branch.

**Comment by stephenbi** [comment]
Submitted a PR from the GCPT3 branch, merge please?

**Comment by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q** [comment]
```
Run calcOutput("Capacity", subtype = "coal2025", round = 6, file = "p47_coalCapCOVID.cs4r")
> Run readSource("GCPT", subtype = "future")
Error in x@.Data[i, j, k] <- value :                                          
  number of items to replace is not a multiple of replacement length
In addition: There were 29 warnings (use warnings() to see them)
> Exit readSource("GCPT", subtype = "future") in 8.92 seconds
Exit calcOutput("Capacity", subtype = "coal2025", round = 6, file = "p47_coalCapCOVID.cs4r") in 9.12 seconds
```
Let me guess: you didn't update `magclass` before testing this on your end, right?

Can be fixed by 
```R
mavgRetAge[getRegions(mavgRetAge)[mavgRetAge == 0],,] <- as.vector(regAvgRetAge[map$RegionCode[which(map$CountryCode %in% getRegions(mavgRetAge)[mavgRetAge == 0])],,])
```

**Comment by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q** [comment]
But then
```
> readGCPT('future')
`summarise()` ungrouping output (override with `.groups` argument)                              
`summarise()` ungrouping output (override with `.groups` argument)                              
`summarise()` ungrouping output (override with `.groups` argument)
`summarise()` ungrouping output (override with `.groups` argument)
`summarise()` ungrouping output (override with `.groups` argument)
`summarise()` ungrouping output (override with `.groups` argument)
NOTE:  - toolCountryFill set missing values for IMPORTANT countries to  0 :                     
NOTE:  --- Afghanistan(AFG)  --- Angola(AGO)  --- Armenia(ARM)  --- Azerbaijan(AZE)  --- Burundi(BDI)  --- Benin(BEN)  --- Burkina Faso(BFA)  --- Bahrain(BHR)  --- Bolivia, Plurinational State of(BOL)  --- Central African Republic(CAF)  --- Switzerland(CHE)  --- Cameroon(CMR)  --- Congo(COG)  --- Costa Rica(CRI)  --- Cuba(CUB)  --- Cyprus(CYP)  --- Algeria(DZA)  --- Ecuador(ECU)  --- Eritrea(ERI)  --- Estonia(EST)  --- Gabon(GAB)  --- Gambia(GMB)  --- Guinea-Bissau(GNB)  --- Haiti(HTI)  --- Iraq(IRQ)  --- Kuwait(KWT)  --- Lebanon(LBN)  --- Liberia(LBR)  --- Libya(LBY)  --- Lesotho(LSO)  --- Lithuania(LTU)  --- Mali(MLI)  --- Mauritania(MRT)  --- Nicaragua(NIC)  --- Norway(NOR)  --- Nepal(NPL)  --- Puerto Rico(PRI)  --- Paraguay(PRY)  --- Palestine, State of(PSE)  --- Qatar(QAT)  --- Rwanda(RWA)  --- Saudi Arabia(SAU)  --- Singapore(SGP)  --- Sierra Leone(SLE)  --- Somalia(SOM)  --- Chad(TCD)  --- Togo(TGO)  --- Turkmenistan(TKM)  --- Timor-Leste(TLS)  --- Trinidad and Tobago(TTO)  --- Tunisia(TUN)  --- Uganda(UGA)  --- Uruguay(URY)  --- Yemen(YEM) 
Error in dimnames(out) <- `*vtmp*` : 'dimnames' applied to non-array
In addition: There were 50 or more warnings (use warnings() to see the first 50)
```
which happens at https://github.com/pik-piam/mrremind/blob/46316d71c40d1861c9f8174639c4bf8a9b3b638b/R/readGCPT.R#L351

No idea what's supposed to happen there.

**Comment by stephenbi** [comment]
Correct you are about not updating... crap. Will pick this up soon.

**Comment by stephenbi** [comment]
Should be actually fixed now in 3d7c16d
